internal/service/flexcluster/model.go

```diff
@@ -1,22 +1,141 @@
-// 	// 	return nil, diagnostics
-// 	// }
-// 	return &admin.FlexCluster{}, nil
-// }
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"go.mongodb.org/atlas-sdk/v20240805004/admin"
+)
+
+func NewTFModel(ctx context.Context, apiResp *admin.FlexClusterDescription20250101) (*TFModel, diag.Diagnostics) {
+	connectionStrings, diags := ConvertConnectionStringsToTF(ctx, apiResp.ConnectionStrings)
+	if diags.HasError() {
+		return nil, diags
+	}
+	backupSettings, diags := ConvertBackupSettingsToTF(ctx, apiResp.BackupSettings)
+	if diags.HasError() {
+		return nil, diags
+	}
+	providerSettings, diags := ConvertProviderSettingsToTF(ctx, apiResp.ProviderSettings)
+	if diags.HasError() {
+		return nil, diags
+	}
+	return &TFModel{
+		ProviderSettings:             *providerSettings,
+		ConnectionStrings:            *connectionStrings,
+		Tags:                         newTFTags(apiResp.Tags),
+		CreateDate:                   types.StringPointerValue(conversion.TimePtrToStringPtr(apiResp.CreateDate)),
+		ProjectId:                    types.StringPointerValue(apiResp.GroupId),
+		Id:                           types.StringPointerValue(apiResp.Id),
+		MongoDbversion:               types.StringPointerValue(apiResp.MongoDBVersion),
+		Name:                         types.StringPointerValue(apiResp.Name),
+		ClusterType:                  types.StringPointerValue(apiResp.ClusterType),
+		StateName:                    types.StringPointerValue(apiResp.StateName),
+		VersionReleaseSystem:         types.StringPointerValue(apiResp.VersionReleaseSystem),
+		BackupSettings:               *backupSettings,
+		TerminationProtectionEnabled: types.BoolPointerValue(apiResp.TerminationProtectionEnabled),
+	}, nil
+}
+
+func NewAtlasCreateReq(ctx context.Context, plan *TFModel) (*admin.FlexClusterDescriptionCreate20250101, diag.Diagnostics) {
+	providerSettings := &TFProviderSettings{}
+	if diags := plan.ProviderSettings.As(ctx, providerSettings, basetypes.ObjectAsOptions{}); diags.HasError() {
+		return nil, diags
+	}
+	return &admin.FlexClusterDescriptionCreate20250101{
+		Name: plan.Name.ValueString(),
+		ProviderSettings: admin.FlexProviderSettingsCreate20250101{
+			BackingProviderName: providerSettings.BackingProviderName.ValueString(),
+			RegionName:          providerSettings.RegionName.ValueString(),
+		},
+		TerminationProtectionEnabled: plan.TerminationProtectionEnabled.ValueBoolPointer(),
+		Tags:                         newResourceTags(ctx, plan.Tags),
+	}, nil
+}
+
+func NewAtlasUpdateReq(ctx context.Context, plan *TFModel) (*admin.FlexClusterDescription20250101, diag.Diagnostics) {
+	updateRequest := &admin.FlexClusterDescription20250101{
```

Comment:
provider settings cannot be updated? Would make sure to confirm API behaviour, from there we can evaluate what error messages/plan modifiers/documentation to incorporate so it is clear to users that updates cannot be performed in certain attributes.

Reply:
The only attributes that can be updated are `tags` and `terminationProtectionEnabled`, see [API docs](https://www.mongodb.com/docs/openapi/preview/?src=https://raw.githubusercontent.com/mongodb/atlas-sdk-go/dev-latest/openapi/atlas-api.yaml#tag/Flex-Clusters/operation/updateFlexCluster).
To avoid making this PR even bigger, I will create a ticket to add error messages/plan modifiers and document which attributes are updatable and which aren't. WDYT?